backend/envconfig/envconfig.go

```diff
@@ -0,0 +1,79 @@
+package envconfig
+
+// TODO(issue#649): move this package into app framework
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/short-d/app/fw"
+)
+
+// EnvConfig parses configuration from environmental variables.
+type EnvConfig struct {
+	environment fw.Environment
+}
+
+// ParseConfigFromEnv retrieves configurations from environmental variables and
+// parse them into the given struct.
+func (e EnvConfig) ParseConfigFromEnv(config interface{}) error {
+	configVal := reflect.ValueOf(config)
+	if configVal.Kind() != reflect.Ptr || configVal.IsNil() {
+		return errors.New("config must be a pointer")
```

Comment:
Should the error message also include that the maybe the config is nil?

Reply:
@arberiii Sure! We can separate the checks!